zilliqa/src/db.rs

```diff
@@ -32,21 +32,23 @@ use crate::{
     transaction::{EvmGas, Log, SignedTransaction, TransactionReceipt},
 };
 
-macro_rules! sqlify_with_bincode {
+macro_rules! sqlify_with_bincode_and_lz4 {
```

Comment:
How much does compressing these values save? Can we measure it and see if it non-negligible?

My worry is that if compressing does shrink the value size a lot, then we should actually be changing how the values are encoded, since there is clearly some waste. And if compressing doesn't change the size, then we shouldn't slow down DB reads and writes by running it.

Or to state my thoughts in another way - Compression isn't 'magic' and it can only remove 'redundant' data. So why can't we just remove that redundant data (if it exists) from the encoded values ourselves?

Reply:
For empty blocks, the compression brings about nett ~1.5% size reduction overall. If we don't want to go overboard with compression, then we can remove them. In that case, we will get ~15% reduction from the WITHOUT ROWID and pragmas.